### ISSUE TITLE
Overlay not drawn near the view edge when map orientation is set

### ISSUE THREAD

**Issue by nullApex** [osmdroid/osmdroid#1462]
## Issue Type

[x] Question
[ ] Bug
[ ] Improvement
[ ] Build system related
[ ] Performance
[ ] Documentation

## Description and/or steps/code to reproduce the problem

When map orientation is set to lets say 45 degrees the polyline overlay isn't drawn near the edges of the screen. Is there a way to fix this?

### If it's a bug, version(s) of android this affects:

All

### Version of osmdroid the issue relates to:

6.1.5

**Comment by monsieurtanuki** [comment]
@nullApex It's surprising but not impossible given all the computations around the polylines.
I could not manage to reproduce this issue with the demo "More Samples / Drawing / Draw a polyline on screen".
Could you please create a little example that reproduces it, then provide us with the Polyline GeoPoint list, the map center, the map zoom level and the map orientation.

**Comment by nullApex** [comment]
@monsieurtanuki Hi, sorry for the delay, here's a sample of the code (it's a mess, but it should be readable).
https://github.com/nullApex/mapTest

**Comment by monsieurtanuki** [comment]
@nullApex Please just copy/paste the Polyline GeoPoint list, the map center, the map zoom level and the map orientation.

**Comment by nullApex** [comment]
Hi, here's a set of polines https://pastebin.com/jKGWaxhb, map center [42.007305, 21.417469], zoom level 19, orientation 323.9

**Comment by monsieurtanuki** [comment]
I managed to reproduce the bug with the following lines:
````java
final List<GeoPoint> list = new ArrayList<>();
list.add(new GeoPoint(42.00643319705,21.41993811227));
list.add(new GeoPoint(42.00663595551,21.4196119));
list.add(new GeoPoint(42.00685516353962,21.41983022208606));
final Polyline polyline = new Polyline();
polyline.setPoints(list);
mMapView.getOverlays().add(polyline);
mMapView.post(new Runnable() {
	@Override
	public void run() {
		final BoundingBox boundingBox = BoundingBox.fromGeoPoints(list);
		mMapView.zoomToBoundingBox(boundingBox, false, 0);
	}
});
````
I had to zoom deeper, rotate the map, then see that the polyline disappeared in the map corner.

I think this is due to a recent change that "optimizes too much": in `DefaultOverlayManager.onDrawHelper` there is now a test about "should this Overlay be displayed given its bounding box" that uses `BoundingBox.overlaps`, but this method doesn't take into account the map orientation.

Therefore:
* in first approach, the `BoundingBox.overlaps` optimization must be removed altogether from `DefaultOverlayManager.onDrawHelper`
* @nullApex You can do that in your own code: just copy `DefaultOverlayManager` into your own class and remove the call to `BoundingBox.overlaps`
* of course we should also remove that `overlaps` check from the main osmdroid code
* in a second step, we should think about implementing that optimization. I guess this will involve `Projection` and the `Overlay.getBounds`

**Comment by nullApex** [comment]
Thanks, that fixed it :D, have a nice day!

**Comment by monsieurtanuki** [comment]
Fixed by #1464

**Comment by monsieurtanuki** [comment]
Created #1465 for the optimization part.
